x/paloma/ante.go

```diff
@@ -78,30 +85,31 @@ func NewVerifyAuthorisedSignatureDecorator(fk types.FeegrantKeeper) VerifyAuthor
 // AnteHandle verifies that the message is signed by at least one signature that has
 // active fee grant from the creator address, IF the message contains metadata.
 func (d VerifyAuthorisedSignatureDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	if simulate {
-		return next(ctx, tx, simulate)
+		return next(sdkCtx, tx, simulate)
 	}
 
 	for _, msg := range tx.GetMsgs() {
 		m, ok := msg.(libmeta.MsgWithMetadata[vtypes.MsgMetadata])
 		if !ok {
-			logger(ctx).Debug(fmt.Sprintf("msg %s does not contain metadata. skipping ownership verification...", proto.MessageName(msg)))
+			liblog.FromSDKLogger(logger(ctx)).Debug(fmt.Sprintf("msg %s does not contain metadata. skipping ownership verification...", proto.MessageName(msg)))
 			continue
 		}
 
 		creator := m.GetMetadata().GetCreator()
-		signers := msg.GetSigners()
+		signers := m.GetMetadata().GetSigners()
```

Comment:
`GetSigners()` was previously defined under `Metadata` now I defined under `MsgWithMetadata` too as suggested above and pushed.

Reply:
Understood your point, here the custom message is not `msg` because it is cosmos-sdk's msg which does not have `GetSigners()` and the custom msg here  is `m.GetMetadata()` and that has `m.GetMetadata().GetSigners()`. Code is upto date with this usage, is anything else need to be added in this PR?  @byte-bandit

Reply:
That's ..... not good. Our entire setup for multiple signing keys is built on top of that interface, and the absence of it would break the ante handler.

Luckily, we don't care about common SDK messages and already skip the handling if the message fails to parse [here](https://github.com/palomachain/paloma/blob/2226e95be60c4972dff44afbbbe7b4cb0ec268c8/x/paloma/ante.go#L87)

Simply using `m.GetMetadata().GetSigners()` won't cut it here though. Use `libmeta.GetSigners(m)`, which performs some additional logic on top of simply reading the signer data.

Reply:
fine, pushed with `libmeta.GetSigners(m)`